### PR TITLE
CosmosDB container management tests

### PR DESCRIPTION
**Pull request by vuqtran88**
## Description
1. Adds tests to test:
   1. Creating CosmosDB container.
   2. Reading CosmosDB container.
   3. Deleting CosmosDB container.
   
1. Refactoring and modifying existing test to be more consistent.
   1. Making sure deleting test database always happen at the end of an exercise.
   2. Removing overlapping tests.
   3. Increasing the CosmosDB emulator partition count.

2. Modifying the `DynamicMethodExecutor` to invoke `Task` based methods synchronously. This helps better test result consistency.

### CODE REVIEW

**Review by JcolemanNR — COMMENTED** [review]
Looks good!

**Review by JcolemanNR — APPROVED** [review]

File: tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/CosmosDB/CosmosDBExerciser.cs [hunk 1]
```diff
@@ -39,61 +40,95 @@ public class CosmosDBExerciser
...
 
-
         [LibraryMethod]
         public static void StartAgent()
         {
             NewRelic.Api.Agent.NewRelic.StartAgent();
             //Get everything started up and time for initial Sample().
             Thread.Sleep(TimeSpan.FromSeconds(10));
         }
-
```
Comment (vuqtran88) [inline tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/CosmosDB/CosmosDBExerciser.cs:92]: We no longer need this as the `DynamicMethodInvoker` can now handle `Task` based method.

File: tests/Agent/IntegrationTests/UnboundedIntegrationTests/CosmosDB/CosmosDBTests.cs [hunk 2]
```diff
@@ -27,11 +27,10 @@ protected CosmosDBTestsBase(TFixture fixture, ITestOutputHelper output)  : base(
             _fixture.SetTimeout(TimeSpan.FromMinutes(2));
 
             _fixture.AddCommand($"CosmosDBExerciser StartAgent");
-            _fixture.AddCommand($"CosmosDBExerciser CreateReadAndDeleteDatabase test_db_{Guid.NewGuid():x}");
-
 
-            _fixture.AddCommand($"CosmosDBExerciser Wait");
```
Comment (vuqtran88) [inline tests/Agent/IntegrationTests/UnboundedIntegrationTests/CosmosDB/CosmosDBTests.cs:33]: We no longer need to wait for an arbitrary amount of time  as the DynamicMethodInvoker can now handle Task based method.

File: tests/Agent/IntegrationTests/Shared/ReflectionHelpers/DynamicMethodExecutor.cs [hunk 3]
```diff
@@ -31,7 +32,13 @@ public void ExecuteDynamicMethod(MethodInfo method, string[] args)
                 }
             }
 
-            method.Invoke(objInst, paramValues.ToArray());
+            var returnValue = method.Invoke(objInst, paramValues.ToArray());
```
Comment (vuqtran88) [inline tests/Agent/IntegrationTests/Shared/ReflectionHelpers/DynamicMethodExecutor.cs:35]: This makes sure the caller wait for `Task` based method invocation to finish before moving on execution other methods.

**Review by JcolemanNR — APPROVED** [review]

File: tests/Agent/UnitTests/Core.UnitTest/Spans/SpanEventMakerTests.cs [hunk 4]
```diff
@@ -70,7 +70,6 @@ public class SpanEventMakerTests
 
         private string _obfuscatedSql;
         private ParsedSqlStatement _parsedSqlStatement;
-        private ParsedSqlStatement _parsedSqlStatementWithEmptyModel;
```
Comment (vuqtran88) [inline tests/Agent/UnitTests/Core.UnitTest/Spans/SpanEventMakerTests.cs:73]: unused variable that i accidentally introduced in the previous PR.

**Review by nr-ahemsath — APPROVED** [review]
LGTM, tests pass locally

**Pull request merged**